### PR TITLE
Create phylum auth token command

### LINKED ISSUE #175: Return current auth token from `auth token`
# Overview
Long-lived authentication tokens are used for the CLI. These tokens are suitable for use in CI/CD. We should make it easier to retrieve these tokens from the `settings.yaml` file by implementing an additional subcommand `phylum auth token`.

```
$ phylum auth token
<UsersAuthToken>
```

# Acceptance Criteria
- [ ] When a user runs `phylum auth token` their long-lived token is printed to stdout.
- [ ] If the user is not authenticated, `auth token` prints a message alerting them to this fact and asking them to authenticate with `phylum auth login`

## Associated User Story
https://www.notion.so/phylum/Token-Retrieval-fd8f391f8f334b30837c5957244f9550

### PR DESCRIPTION
**Pull request by mdx97**
# Overview

This PR implements the `phylum auth token` command to print out the user's current long-lived token that is stored in their `settings.yaml` file.

I will also be creating a companion PR in the `phylum-types` repository to implement `Display` for `RefreshToken`.
## Checklist

- [x] Does this PR have an associated issue?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [ ] Have you created sufficient tests?
- [x] Have you updated all affected documentation?

## Issue
closes #175

### CODE REVIEW

**Comment by mdx97** [comment]
I wasn't able to figure out if there was a correct place / pattern for writing tests for this specific command. It seems that the `phylum auth status` command doesn't really have associated tests, so I wasn't sure if they were necessary / expected here.  If they are, let me know and I will write some!

File: cli/src/bin/phylum/commands/auth.rs
Comment (janasheehan) [inline cli/src/bin/phylum/commands/auth.rs:46]: am just wondering if we could phrase this warning a bit better - like `Missing authentication, please use  phylum auth login` or some such

Comment (louislang) [reply cli/src/bin/phylum/commands/auth.rs:46]: I don't have strong opinions on the warning, both options seem clear enough to me. I think we might want to delineate where the command starts though. For example: 

```
...you can login with `phylum auth login`
```

**Comment by louislang** [comment]
> I wasn't able to figure out if there was a correct place / pattern for writing tests for this specific command. It seems that the `phylum auth status` (which is similar to what I've implemented here) command doesn't really have associated tests, so I wasn't sure if they were necessary / expected here. If they are, let me know and I will write some!

Great addition, @mdx97! More tests are always better, and definitely appreciate improving our code coverage where possible.

File: cli/src/bin/phylum/commands/auth.rs
Comment (janasheehan) [reply cli/src/bin/phylum/commands/auth.rs:46]: yeah, it's totally not important - just the mixture of third person, and second person feels a bit off - since the warning is for the `user`

**Comment by kylewillmon** [comment]
The updates to `phylum-types` should be merged to master in phylum-dev/phylum-types#17 (which is waiting on some internal API changes to be completed).

I will re-run the checks and merge this PR once that is done.

**Review by kylewillmon — DISMISSED** [review]

**Comment by kylewillmon** [comment]
I made changes to this branch resolve @janasheehan's [concerns about wording](https://github.com/phylum-dev/cli/pull/217#discussion_r836889015) and to update the branch after the #218 merge, so I'm adding @maxrake as a reviewer to avoid being both reviewer and now partial author of this PR.

**Review by janasheehan — DISMISSED** [review]

**Review by maxrake — DISMISSED** [review]
@mdx97, thanks for contributing!

LGTM

**Comment by kylewillmon** [comment]
We are now tracking the `development` branch of `phylum-types`, which unblocks this PR

**Review by maxrake — APPROVED** [review]
Checks pass...code changes (merge from `development` branch) look fine...LGTM.

**Pull request merged**
